lib/rules/require-named-capture-group.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * @fileoverview Rule to enforce requiring named capture groups in regular expression.
+ * @author Pig Fang <https://github.com/g-plane>
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const regexpp = require("regexpp");
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+const parser = new regexpp.RegExpParser();
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        type: "suggestion",
+
+        docs: {
+            description: "enforce using named capture group in regular expression",
+            category: "Best Practices",
+            recommended: false,
+            url: "https://eslint.org/docs/rules/require-named-capture-group"
+        },
+
+        schema: [],
+
+        messages: {
+            required: "Capture group '{{group}}' in regular expression should be named."
+        }
+    },
+
+    create(context) {
+
+        /**
+         * Function to check regular expression.
+         *
+         * @param {string} regex The regular expression to be check.
+         * @param {ASTNode} node AST node which contains regular expression.
+         * @returns {void}
+         */
+        function checkRegex(regex, node) {
+            const ast = parser.parsePattern(regex);
```

Comment:
Giving `uFlag` argument is important because `u` flag changes the regular expression syntax massively. If `u` flag value is different, it can throw an unexpected syntax error.

And `parsePattern` method can throw syntax errors, please wrap with a `try` statement.

Reply:
Should we need to report a warning/error if syntax errors threw?

Reply:
I think that this rule should ignore syntax errors because the purpose of this rule is not catching syntax error. A rule should focus on one thing :)